src/components/organisms/NavBarSchedule/NavBarSchedule.tsx

```diff
@@ -0,0 +1,155 @@
+  useCallback,
+} from "react";
+import {
+  addDays,
+  startOfToday,
+  format,
+  getUnixTime,
+  fromUnixTime,
+} from "date-fns";
+import { groupBy, range } from "lodash";
+import classNames from "classnames";
+
+import { SCHEDULE_SHOW_DAYS_AHEAD } from "settings";
+
+import { useRelatedVenues } from "hooks/useRelatedVenues";
+import { useVenueId } from "hooks/useVenueId";
+import { useUser } from "hooks/useUser";
+import { useVenueEvents } from "hooks/events";
+
+import {
+  PersonalizedVenueEvent,
+  VenueLocation,
+  LocatedEvents,
+  VenueEvent,
+} from "types/venues";
+
+import { Schedule } from "components/molecules/Schedule";
+import { ScheduleVenueDescription } from "components/molecules/ScheduleVenueDescription";
+
+import {
+  buildLocationString,
+  extractLocation,
+  prepareForSchedule,
+} from "./utils";
+import { isEventWithinDate } from "utils/event";
+import { WithVenueId } from "utils/id";
+
+import "./NavBarSchedule.scss";
+
+interface NavBarScheduleProps {
+  isVisible?: boolean;
+}
+
+const emptyRelatedEvents: WithVenueId<VenueEvent>[] = [];
+
+export interface ScheduleDay {
+  isToday: boolean;
+  dayStartUtcSeconds: number;
+  locatedEvents: LocatedEvents[];
+  personalEvents: PersonalizedVenueEvent[];
+}
+
+export const emptyPersonalizedSchedule = {};
+
+export const NavBarSchedule: FC<NavBarScheduleProps> = ({ isVisible }) => {
+  const venueId = useVenueId();
+  const { userWithId } = useUser();
+  const userEventIds =
+    userWithId?.myPersonalizedSchedule ?? emptyPersonalizedSchedule;
+
+  const { isLoading, relatedVenues, relatedVenueIds } = useRelatedVenues({
+    currentVenueId: venueId,
+  });
+
+  const {
+    isEventsLoading,
+    events: relatedVenueEvents = emptyRelatedEvents,
+  } = useVenueEvents({
+    venueIds: relatedVenueIds,
+  });
+
+  const isLoadingSchedule = isLoading || isEventsLoading;
+
+  const [selectedDayIndex, setSelectedDayIndex] = useState(0);
+
+  const weekdays = useMemo(() => {
+    const today = startOfToday();
+
+    return range(0, SCHEDULE_SHOW_DAYS_AHEAD).map((dayIndex) => {
+      const day = addDays(today, dayIndex);
+      const classes = classNames("NavBarSchedule__weekday", {
+        "NavBarSchedule__weekday--active": dayIndex === selectedDayIndex,
+      });
+
+      const onWeekdayClick: MouseEventHandler<HTMLLIElement> = () => {
+        setSelectedDayIndex(dayIndex);
+      };
+
+      return (
+        <li
+          key={day.toISOString()}
+          className={classes}
+          onClick={onWeekdayClick}
+        >
+          {dayIndex === 0 ? "Today" : format(day, "E")}
+        </li>
+      );
+    });
+  }, [selectedDayIndex]);
+
+  const getEventLocation = useCallback(
+    (locString: string): VenueLocation => {
+      const [venueId, roomTitle = ""] = extractLocation(locString);
+      const venueTitle = relatedVenues.find((venue) => venue.id === venueId)
+        ?.name;
+      return { venueId, roomTitle, venueTitle };
+    },
+    [relatedVenues]
+  );
```

Comment:
Should we also allow `roomTitle` to actually be `undefined` here rather than defaulting it to an empty string? (like we did with `venueTitle`)

Reply:
removed `""`: https://github.com/sparkletown/sparkle/pull/1426/commits/6ff0df273fef51c6b69a313b4be36fa82d94a699